### PR TITLE
feat(sql-execution-plan): add errorCode for intelligent analysis of sql execution plan

### PR DESCRIPTION
**Pull request by PeachThinking**
#### What this PR does / why we need it:
add errorCode for intelligent analysis of sql execution plan
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-core/src/main/java/com/oceanbase/odc/core/shared/constant/ErrorCodes.java [hunk 1]
```diff
@@ -257,6 +257,7 @@ public enum ErrorCodes implements ErrorCode {
     ObFullLinkTraceNotSupported,
     ObFullLinkTraceNotEnabled,
     ObQueryProfileNotSupported,
+    IntelligentAnalysisOfExecutionPlanFailed,
```
Comment (yizhouxw) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/shared/constant/ErrorCodes.java:260]: how about use 'Copilot' as AI module prefix ?

e.g. CopilotAnalyzeExecutoinPlanFailed

**Review by yizhouxw — COMMENTED** [review]
Let

File: server/odc-core/src/main/java/com/oceanbase/odc/core/shared/constant/ErrorCodes.java (hunk 1 above)
Comment (PeachThinking) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/shared/constant/ErrorCodes.java:260]: good idea～

**Review by Jane201510 — DISMISSED** [review]
agree

File: server/odc-core/src/main/resources/i18n/ErrorMessages.properties [hunk 2]
```diff
@@ -142,6 +142,7 @@ com.oceanbase.odc.ErrorCodes.ObCreatePlRunPackageFailed=Failed to create the PL
 com.oceanbase.odc.ErrorCodes.ObGetExecuteDetailFailed=Failed to fetch the execution details. Details: {0}.
 com.oceanbase.odc.ErrorCodes.ObGetPlanExplainFailed=Failed to fetch the execution plan due to possible syntax error or invalid DDL statement. Details: {0}.
 com.oceanbase.odc.ErrorCodes.ObGetPlanExplainEmpty=Unable to fetch the execution plan due to possible syntax error, invalid DDL statement, or unsupported command. Details: {0}.
+com.oceanbase.odc.ErrorCodes.IntelligentAnalysisOfExecutionPlanFailed=SQL execution plan intelligent analysis failed, error details: {0}.
```
Comment (JessieWuJiexi) [inline server/odc-core/src/main/resources/i18n/ErrorMessages.properties]: Failed to perform intelligent analysis of the SQL execution plan. Details: {0}.

Comment (PeachThinking) [reply server/odc-core/src/main/resources/i18n/ErrorMessages.properties]: ok～

**Review by JessieWuJiexi — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by Jane201510 — APPROVED** [review]
agree

**Pull request merged**